gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/QueryStage.java

```diff
@@ -373,8 +373,9 @@ public String toString() {
 
   @Override
   public int hashCode() {
```

Comment:
What policy should we follow with `final` and `equals`/`hashCode` methods?
Do we want all `equals()` and `hashCode()` to be final? Or just for functional classes and configurations? (where bad subclassing would be more painful). Once we decide this I can take care of making everything more consistent.